pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -1170,29 +1195,31 @@ func applyUpstreamTLSSettings(env *model.Environment, cluster *apiv2.Cluster, tl
 
 	// convert to transport socket matcher if the mode was auto detected
 	if tls.Mode == networking.TLSSettings_ISTIO_MUTUAL && mtlsCtxType == autoDetected && util.IsIstioVersionGE14(proxy) {
-		tlsContext, err := ptypes.MarshalAny(cluster.TlsContext)
-		cluster.TlsContext = nil
+		istioTLSContext, err := ptypes.MarshalAny(cluster.TlsContext)
 		if err != nil {
-			log.Errorf("error marshaling tls context to transport_socket config for cluster %s, err=%v",
+			log.Errorf("error marshaling istio tls context to transport_socket config for cluster %s, err=%v",
 				cluster.Name, err)
 			return // no tls context for the cluster
 		}
+		cluster.TlsContext = nil
+
 		cluster.TransportSocketMatches = []*apiv2.Cluster_TransportSocketMatch{
 			{
-				Name: "mtls",
+				Name: model.IstioMutualTLSModeLabel,
```

Comment:
Out of curiosity, are there any upcoming uses/features/design plans for TransportSocketMatches in upstream envoy where `istio` would not be an appropriate name?  i.e. if we as istio developers decide to append additional TransportSocketMatches to cover additional features, `istio` may not be appropriate since we would be appending more than one Istio-themed socket match.  My assumption is these cluster settings will remain pretty invisible to users so changing names down the line shouldn't be breaking from a customer standpoint correct?

Reply:
So this is just name of object and not the type. We have to provide some name and this is mostly used for stats or just logging.